client/src/components/LocationEntry/container.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { retrieveLocationById } from "Services/LocationService";
+import { retrievePlantsByLocationId } from "Services/PlantService";
+
+export default Wrapped =>
+  class extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        location: "",
+        plants: ""
+      };
+    }
+
+    componentDidMount() {
+      retrieveLocationById(this.props.id).then(response => {
```

Comment:
You can actually use an axios approach to kick these calls off at the same time and to call a method when they both finish.

 axios.all([retrieveLocationById(this.props.id), retrievePlantsByLocationId(this.props.id])
        .then(axios.spread((location, plants) => {
            this.setState({
              location: response.data[0],
              plants: response.data
            });
          }
        }))

Reply:
ooh good tip, thanks! :)